Firmware/includes/controller.c

```diff
@@ -34,47 +34,51 @@ void control_init(void)
 	DDRD |= (1<< PD4);
 	w = config.default_temp;
 }
-
-void control(void)
+uint8_t t=0;
+void control(int16_t temp)
 {
-	// XXX test this
-	int16_t temp = tip_get_temp();
+	static int16_t old_temp = 0;
 	int16_t e = w - temp; 
-	//display_fixed_point((uint16_t)(config.pid_p * 10),-1);
-	   //guarding preventing esum over/underflows
+	
+	//guarding
 	if(esum >= (int32_t)(0x3fff - e)){
 		esum = 0x3fff;
 	} else if(esum <= (int32_t)(-0x3fff + e)){
 		esum = (-0x3fff);
 	}else{
-			esum += e;
+		esum += e;
 	}
-
-	int16_t p = (uint32_t)(config.pid_p*e)/(uint32_t)128;
-	int16_t i = (int32_t)((int32_t)config.pid_i*(int32_t)Ta*(int32_t)esum)/(uint32_t)128;
-	int16_t d = (int32_t)(config.pid_d/Ta*(e - eold))/(uint32_t)128;
+	
+	int16_t p = ((int32_t)((int32_t)config.pid_p*(int32_t)e))/(int32_t)128;
+	int16_t i = 0;//(int32_t)((int32_t)config.pid_i*(int32_t)Ta*(int32_t)esum)/(uint32_t)128;
+	int16_t d = 0;//(int32_t)(config.pid_d/Ta*(e - eold))/(uint32_t)128;
 	int16_t y = (int32_t)((int32_t)p + (int32_t)i + (int32_t)d);
 	eold = e;
-
-	   //guarding
+	
+	
+	//guarding
 	if(y > 255){
```

Comment:
Ok that is ulgy!!! 
Let us first try to set good PID parameters, that is normally the way to avoid such behavior.
If that does not work, we can reconsider this option